src/System.Numerics.Experimental/Half.cs

```diff
@@ -353,5 +356,402 @@ public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan
+
+            uint alignedSig = (sig | 0x0400) << shiftDist;
+            return alignedSig >> 10;
+        }
+
+        // TODO: confirm behaviours
+        public static explicit operator short(Half h)
+        {
+            return (short) (int) h;
+        }
+
+        public static explicit operator ushort(Half h)
+        {
+            return (ushort)(short)(int)h;
+        }
+
+        public static explicit operator byte(Half h)
+        {
+            return (byte)(sbyte)(int)h;
+        }
+
+        public static explicit operator sbyte(Half h)
+        {
+            return (sbyte)(int)h;
+        } // TODO: not sure what should happen here
+
+        public static implicit operator float(Half h)
+        {
+            bool sign = IsNegative(h);
+            int exp = h.Exponent;
+            uint sig = h.Significand;
+
+            if (exp == MaxExponent)
+            {
+                if (sig != 0)
+                    return CreateSingleNaN(sign, (ulong) sig << 54);
+                return sign ? float.NegativeInfinity : float.PositiveInfinity;
+            }
+
+            if (exp == 0)
+            {
+                if (sig == 0)
+                    return BitConverter.Int32BitsToSingle(unchecked((int)(sign ? SingleSignMask : 0))); // Positive / Negative zero
+                (exp, sig) = NormSubnormalF16Sig(sig);
+                exp -= 1;
+            }
+
+            return CreateSingle(sign, (byte)(exp + 0x70), sig << 13);
+        }
+
+        public static implicit operator double(Half h)
+        {
+            bool sign = IsNegative(h);
+            int exp = h.Exponent;
+            uint sig = h.Significand;
+
+            if (exp == MaxExponent)
+            {
+                if (sig != 0)
+                    return CreateDoubleNaN(sign, (ulong) sig << 54);
+                return sign ? double.NegativeInfinity : double.PositiveInfinity;
+            }
+
+            if (exp == 0)
+            {
+                if (sig == 0)
+                    return BitConverter.Int64BitsToDouble(unchecked((long) (sign ? DoubleSignMask : 0))); // Positive / Negative zero
+                (exp, sig) = NormSubnormalF16Sig(sig);
+                exp -= 1;
+            }
+
+            return CreateDouble(sign, (ushort)(exp + 0x3F0), sig << 42);
+        }
+
+        // IEEE 754 specifies NaNs to be propagated
+        public static Half operator -(Half h)
+        {
+            return IsNaN(h) ? h : new Half((ushort)(h.m_value ^ SignMask));
+        }
+
+        public static Half operator +(Half h)
+        {
+            return h;
+        }
+
+        #region Utilities
+
+        // TODO: Worth bringing the `ShortShiftRightJam`? looks like some perf difference only
+        // If any bits are lost by shifting, "jam" them into the LSB.
+        // if dist > bit count, Will be 1 or 0  depending on i 
+        // (unlike bitwise operators that masks the lower 5 bits)
+        private static uint ShiftRightJam(uint i, int dist)
```

Comment:
note: this basically looks like a rotate function. but with different functionality if the distance is larger than the source bit count.